src/tool/xmeta/cpp/models/delegate_model.h

```diff
@@ -4,17 +4,19 @@
 #include <optional>
 #include <vector>
 
-#include "base_model.h"
-#include "model_types.h"
+#include "namespace_member_model.h"
 #include "formal_parameter_model.h"
+#include "model_types.h"
 
 namespace xlang::xmeta
 {
-    struct delegate_model : base_model
+    struct namespace_body_model;
```

Comment:
since you have the header above, this forward decl is superfluous, right?

Reply:
We include the namespace_member_model, which generalizes a namespace member. Namespace bodies are defined in `namespace_model.h`. Namespace members contain reference to their containing body, however we don't need to know how it's defined hence the forward declare.